drivers/random/random_nrf5.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2017 Exati Tecnologia Ltda.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <random.h>
+#include <atomic.h>
+#include <nrf_rng.h>
+
+struct random_nrf5_dev_data {
+	atomic_t user_count;
+};
+
+#define DEV_DATA(dev) \
+	((struct random_nrf5_dev_data *)(dev)->driver_data)
+
+static inline s16_t random_nrf5_get_u8(void)
+{
+	u8_t value;
+
+	while (!nrf_rng_event_get(NRF_RNG_EVENT_VALRDY)) {
+		__WFE();
```

Comment:
This is not going to work properly. The VALRDY event itself won't wake up the core. You would need to generate an interrupt for this to happen (see for instance [here](http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.dui0553a/CHDEAAHJ.html)). VALRDY is a peripheral event, not to be confused with the internal event register in the core that influences WFE.